components/Builder/Builder.tsx

```diff
@@ -120,13 +134,22 @@ const Builder = () => {
 
 export default Builder;
 
+function useBuilder() {
```

Comment:
nit: what't the point of this function?

Reply:
At least in Across, we've typically tried to split out the view and business logic as much as possible, preferring to put useState variables, all effects (etc) in tiny hooks rather than the component itself.